bpfd/src/multiprog/tc.rs

```diff
@@ -110,19 +112,72 @@ impl TcDispatcher {
             loader: Some(loader),
         };
         dispatcher.attach_extensions(&mut extensions).await?;
-        dispatcher.attach(old_dispatcher)?;
+        dispatcher.attach(old_dispatcher).await?;
         dispatcher.save()?;
         Ok(dispatcher)
     }
 
-    fn attach(&mut self, old_dispatcher: Option<Dispatcher>) -> Result<(), BpfdError> {
+    /// has_qdisc returns true if the qdisc_name is found on the if_index.
+    async fn has_qdisc(qdisc_name: String, if_index: i32) -> Result<bool, anyhow::Error> {
+        let (connection, handle, _) = rtnetlink::new_connection().unwrap();
+        tokio::spawn(connection);
+
+        let mut qdiscs = handle.qdisc().get().execute();
+        while let Some(qdisc_message) = qdiscs.try_next().await? {
+            if qdisc_message.header.index == if_index {
+                debug!(
+                    "Qdisc: {:?} - Parent: {:?} - Handle: {:?} - Kind: {:?}",
+                    qdisc_message.header.index,
+                    qdisc_message.header.parent,
+                    qdisc_message.header.handle,
+                    qdisc_message.header.info
+                );
+                for nla in qdisc_message.nlas {
+                    if let Nla::Kind(nla_kind) = nla {
+                        if nla_kind == qdisc_name {
+                            return Ok(true);
+                        }
+                    }
+                }
+            }
+        }
+        Ok(false)
```

Comment:
I'm not necessarily saying to do this, but it seemed like we could use the try_filter() method to skip the messages we don't care about as an option to what you are doing on lines 126 & 127.  So, I played around with it and came up with the following.  What the following code does is skip everything except for what we are looking for.  So, after that's done, if anything is left, it's the qdisc we are looking for on the if_index provided.

I'd like to get @dave-tucker's opinion on two things.  First of all, the pattern @navarrothiago used seems to follow those in the rtnetlink examples (https://github.com/rust-netlink/rtnetlink/tree/main/examples).  However, the example of yours that you pointed us to seems a bit different and lower level.  Are you okay with this?  Also, do you have a preference for either Thiago's option or the one below?

```suggestion
        if (handle
            .qdisc()
            .get()
            .execute()
            .try_filter(|msg| {
                future::ready(
                    msg.header.index == if_index
                        && msg.nlas.contains(&Nla::Kind(qdisc_name.clone())),
                )
            })
            .try_next()
            .await?)
            .is_some()
        {
            return Ok(true);
        }        
        Ok(false)
```